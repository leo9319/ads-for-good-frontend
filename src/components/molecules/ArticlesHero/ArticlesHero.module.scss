@import '../../../styles/breakpoints.scss';
@import '../../../styles/colours';

.articlesWrapper {
  padding: var(--spacing-2xl) 0;
  box-sizing: border-box;
  background-color: $color-neutral-50;

  .articlesContainerWidth {
    width: 60.5%;
    margin: 0 auto;

    .articlesBannerSection {
      .articlesImage {
        min-width: 100%;
        max-width: 100%;
        width: 870px;
      }
    }
  }
  .articlesFullWidth {
    width: 100%;

    .articlesBannerSection {
      .articlesBannerSectionImageWrapper {
        width: 50%;

        .articlesImage {
          min-width: 100%;
          max-width: 100%;
        }
      }

      .articlesContentSection {
        width: 50%;
      }
    }
  }

  .articlesContentSection {
    padding: 0 var(--spacing-md);
  }

  .articlesTitle {
    color: $color-neutral-950;
    margin: var(--spacing-md) 0;
  }

  .articlesDescription {
    color: $color-neutral-600;
    margin-bottom: var(--spacing-lg);
  }

  .articlesAuthorName {
    color: $color-neutral-600;
    margin-bottom: var(--spacing-2xs);

    b {
      color: $color-neutral-950;
      font-weight: var(--typography-font-weight-regular);
    }
  }
}

@media (max-width: $xl-min) {
  .articlesWrapper {
    .articlesTitle {
      width: 96.337% !important;
    }

    .articlesDescription {
      width: 96.338% !important;
    }
  }
}

@media (max-width: $lg-min) {
  .articlesWrapper {
    .articlesTitle {
      width: 83.153% !important;
    }

    .articlesDescription {
      width: 83.153% !important;
    }
  }
}

@media (max-width: $md-min) {
  .articlesWrapper {
    .articlesContainerWidth {
      width: 89.563%;

      .articlesBannerSection {
        .articlesImage {
          width: 784px;
        }
      }
    }

    .articlesTitle {
      font-size: var(--typography-font-size-7);
      line-height: var(--typography-font-size-8);
      width: 83.153% !important;
    }

    .articlesDescription {
      font-size: var(--typography-font-size-3);
      width: 83.153% !important;
    }
  }
}

@media (max-width: $sm-min) {
  .articlesWrapper {
    .articlesContainerWidth {
      width: 90.2%;

      .articlesBannerSection {
        .articlesImage {
          width: 688px;
        }
      }
    }

    .articlesTitle {
      width: 96.863% !important;
    }

    .articlesDescription {
      width: 96.863% !important;
    }

    .articlesFullWidth {
      .articlesBannerSection {
        .articlesBannerSectionImageWrapper {
          width: 100%;
        }

        .articlesContentSection {
          width: 100%;
        }
      }
    }
  }
}

@media (max-width: $xs-min) {
  .articlesWrapper {
    padding-top: 0;

    .articlesContainerWidth {
      width: 100%;

      .articlesBannerSection {
        .articlesImage {
          width: inherit;
        }
      }
    }

    .articlesTitle {
      font-size: var(--typography-font-size-6);
    }

    .articlesDescription {
      width: 100% !important;
    }
  }
}
