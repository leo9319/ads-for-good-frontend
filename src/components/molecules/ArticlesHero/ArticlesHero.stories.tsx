import React from 'react';
import Article from './ArticlesHero';
import { Meta, StoryObj } from '@storybook/react/*';
import FallbackImage from '@assets/images/FallbackImage.svg';

const meta = {
  title: 'Library/Molecules/Article',
  component: Article,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    imagePosition: {
      options: ['top', 'right', 'left'],
      control: { type: 'select' },
    },
    imageUrl: {
      control: 'text',
    },
    imageAltText: {
      control: 'text',
    },
    imageFallback: {
      control: 'text',
    },
    articleTitle: {
      control: 'text',
    },
    articleDescription: {
      control: 'text',
    },
    authorName: {
      control: 'text',
    },
    authoredDate: {
      control: 'text',
    },
    className: {
      control: 'text',
    },
  },
  decorators: [
    Story => (
      <div>
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof Article>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ArticleHeroContent: Story = {
  args: {
    imagePosition: 'top',
    imageUrl:
      'https://stg-beta.worldvision.ca/adobe/dynamicmedia/deliver/dm-aid--3a8bc598-805b-426b-aad6-d88cfe303ebb/childsponsorship.jpeg',
    imageAltText: 'dummy alt',
    imageFallback: FallbackImage,
    articleTitle:
      'An education interrupted: A Venezuelan’s story of escaping oppression',
    articleDescription:
      'In 2023, we helped 16.5+ million of the world’s most vulnerable children and families overcome poverty. ',
    authorName: 'Melanie Ramos',
    authoredDate: '2025-01-11',
    className: '',
  },
};
