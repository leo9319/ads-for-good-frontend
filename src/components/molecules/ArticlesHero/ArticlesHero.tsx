import React from 'react';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import Flex from '@radix-styles/atoms/Flex';
import Text from '@radix-styles/atoms/Text';
import { Heading } from '@radix-styles/atoms/Heading';
import Image from '@components/atoms/Image';

import styles from './ArticlesHero.module.scss';
import { formatDate } from '@utils/common/date';

export interface ArticlesHeroProps {
  /**
   * Image position for the Articles
   * @property
   * @default 'top'
   */
  imagePosition?: 'top' | 'right' | 'left';
  /**
   * Image url for the Articles
   * @property
   */
  imageUrl?: string;
  /**
   * Image alt text
   * @property
   */
  imageAltText?: string;
  /**
   * Fallback image url for the Articles
   * @property
   */
  imageFallback?: string;
  /**
   * Title for the Articles
   * @property
   */
  articleTitle?: string;
  /**
   * Description for the Articles
   * @property
   */
  articleDescription?: string;
  /**
   * Author name for the Articles
   * @property
   */
  authorName?: string;
  /**
   * Authored date for the Articles
   * @property
   */
  authoredDate?: string;
  /**
   * Class name for Articles container
   * @property
   */
  className?: string;
}
export const ArticlesHero = ({
  imagePosition = 'top',
  imageUrl,
  imageAltText,
  imageFallback,
  articleTitle,
  articleDescription,
  authorName,
  authoredDate,
  className,
}: ArticlesHeroProps) => {
  const isTop = imagePosition == 'top';
  const row = imagePosition == 'left' ? 'row' : 'row-reverse';
  const direction = isTop ? 'column' : row;
  const classes = getModuleClasses(className?.trim(), styles);
  const containerWidth = isTop
    ? classNames(styles.articlesContainerWidth)
    : classNames(styles.articlesFullWidth);

  const formattedDate = formatDate(authoredDate);

  return (
    <div className={classNames(styles.articlesWrapper, classes)}>
      <div className={containerWidth}>
        <Flex
          align="center"
          direction={direction}
          wrap="wrap"
          className={classNames(styles.articlesBannerSection)}
        >
          <Flex
            className={classNames(styles.articlesBannerSectionImageWrapper)}
          >
            <Image
              minWidth="100%"
              maxWidth="100%"
              src={imageUrl ?? ''}
              alt={imageAltText ?? ''}
              fallbackSrc={imageFallback}
              className={classNames(styles.articlesImage)}
            />
          </Flex>
          <Flex
            align="center"
            direction="column"
            wrap="wrap"
            className={classNames(styles.articlesContentSection)}
          >
            {articleTitle && (
              <Heading
                size="9"
                weight="semibold"
                align="center"
                className={classNames(styles.articlesTitle)}
              >
                {articleTitle}
              </Heading>
            )}
            {articleDescription && (
              <Text
                size="4"
                align="center"
                className={classNames(styles.articlesDescription)}
              >
                {articleDescription}
              </Text>
            )}
            {authorName && (
              <Text
                size="1"
                align="center"
                className={classNames(styles.articlesAuthorName)}
              >
                Written by <b>{authorName}</b>
              </Text>
            )}
            {formattedDate && (
              <Text
                size="1"
                align="center"
                className={classNames(styles.articlesAuthorName)}
              >
                on <b>{formattedDate}</b>
              </Text>
            )}
          </Flex>
        </Flex>
      </div>
    </div>
  );
};

export default ArticlesHero;
