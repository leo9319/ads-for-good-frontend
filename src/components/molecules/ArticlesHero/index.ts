export { ArticlesHero } from './ArticlesHero';
export { default } from './ArticlesHero';
