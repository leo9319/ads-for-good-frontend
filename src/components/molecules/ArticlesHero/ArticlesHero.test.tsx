import React from 'react';
import ArticlesHero from './ArticlesHero';
import { render } from '@testing-library/react';

describe('Article', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <ArticlesHero
        imagePosition="top"
        imageUrl="https://stg-beta.worldvision.ca/adobe/dynamicmedia/deliver/dm-aid--3a8bc598-805b-426b-aad6-d88cfe303ebb/childsponsorship.jpeg"
        imageAltText="dummy alt"
        imageFallback=""
        articleTitle="An education interrupted: A Venezuelan’s story of escaping oppression"
        articleDescription="In 2023, we helped 16.5+ million of the world’s most vulnerable children and families overcome poverty."
        authorName="Melanie Ramos"
        authoredDate="2025-01-11"
        className=""
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
