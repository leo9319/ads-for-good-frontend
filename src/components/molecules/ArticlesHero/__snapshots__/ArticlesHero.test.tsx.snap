// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render successfully 1`] = `
<body>
  <div>
    <div
      class="articlesWrapper"
    >
      <div
        class="articlesContainerWidth"
      >
        <div
          class="rt-Flex rt-r-fd-column rt-r-ai-center rt-r-fw-wrap articlesBannerSection"
        >
          <div
            class="rt-Flex articlesBannerSectionImageWrapper"
          >
            <img
              alt="dummy alt"
              aria-hidden="true"
              class="rt-Skeleton imgWrapper articlesImage"
              inert=""
              src="https://stg-beta.worldvision.ca/adobe/dynamicmedia/deliver/dm-aid--3a8bc598-805b-426b-aad6-d88cfe303ebb/childsponsorship.jpeg"
              style="--imgMinWidth: 100%; --imgMaxWidth: 100%;"
              tabindex="-1"
            />
          </div>
          <div
            class="rt-Flex rt-r-fd-column rt-r-ai-center rt-r-fw-wrap articlesContentSection"
          >
            <h1
              class="rt-Heading rt-r-size-6 rt-r-ta-center headingContainer mocked-class-name typography-font-family-default articlesTitle"
              id=""
            >
              An education interrupted: A Venezuelan’s story of escaping oppression
            </h1>
            <span
              class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default articlesDescription"
              id=""
            >
              In 2023, we helped 16.5+ million of the world’s most vulnerable children and families overcome poverty.
            </span>
            <span
              class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default articlesAuthorName"
            >
              Written by 
              <b>
                Melanie Ramos
              </b>
            </span>
            <span
              class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default articlesAuthorName"
            >
              on 
              <b>
                Jan 11, 2025
              </b>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
